install.sh

```diff
@@ -1,62 +1,248 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-if [[ -e /etc/debian_version ]] || [[ -e /etc/fedora-release ]] || [[ -e  /etc/centos-release ]] || [[ -e /etc/arch-release ]]; then
-	:
-else
-	echo "Looks like you aren't running this installer on a Debian, Ubuntu, Fedora, CentOS, or Arch Linux system."
-	exit 1
-fi
+clear
+# Clear screen before running any commands
 
+REQUIREMENTS_INSTALLED=0
+# Implemented in install_MAIN
+DEPS_ONLY=0
+# Implemented at bottom of file
+PDEPS_ONLY=0
+# Implemented at bottom of file
+ASSUME_YES=0
+# Implemented in confirm function
 
-command -v jq >/dev/null 2>&1 && { JQ=0; }
-#install python
-if ! command -v python3 &> /dev/null; then
-	if [[ -e /etc/debian_version ]]; then
-		sudo apt install python3
-	elif [[ -e /etc/fedora-release ]]; then
-		sudo dnf install python3
-	elif [[ -e /etc/centos-release ]]; then
-		sudo yum install -y python3
-	elif [[ -e /etc/arch-release ]]; then
-		sudo pacman -S python3
-	fi
-else
-	echo "Skipping installing python, as python is preinstalled."
-fi
-#install tkinter, a dependency
-#install jq, needed to get version number
-if [[ -e /etc/debian_version ]]; then
-	sudo apt install python3-tk jq python3-pip
-elif [[ -e /etc/fedora-release ]]; then
-	sudo dnf install python3-tkinter jq python3-pip
-elif [[ -e /etc/centos-release ]]; then
-	rpm -q epel-release &> /dev/null || EPEL=0 
-	sudo yum install -y python3-tkinter epel-release python3-pip
-	sudo yum install -y jq
-	[[ $EPEL -eq 0 ]] && sudo yum remove -y epel-release
-elif [[ -e /etc/arch-release ]]; then
-	sudo pacman -S --needed tk jq
-fi
+print_usage() {
+   # Display Usage
+   echo "Usage: $0 [-h] [-d]"
+   echo
+   echo "Installer script for The YouTube Spammer Purge application."
+   echo
+   echo "Options:"
+   echo "    -y     Assume yes: Skip confirmation prompts."
+   echo "    -d     Only install OS-Specific dependencies."
+   echo "    -c     Skip installing OS-Specific dependencies."
+   echo "           This could be useful to those who have unsupported systems"
+   echo "    -p     Only install Python dependencies"
+   echo "    -h     Print this Help."
+   echo
+}
+
+# Get the options
+while getopts ":hdcyp" option; do
+	case $option in
+		h)  # display Help
+			print_usage
+			exit;;
+		d)  # Install dependencies only
+			DEPS_ONLY=1;;
+		c)  # Skip installing dependencies
+			REQUIREMENTS_INSTALLED=1;;
+		p)  # Install Python dependencies only
+			PDEPS_ONLY=1;;
+		y)  # Assume YES
+			ASSUME_YES=1;;
+		\?) # Invalid option
+			echo "Error: Invalid option. See option -h for help."
+			exit 1 ;;
+	esac
+done
+
+# Credit to https://stackoverflow.com/questions/29436275/how-to-prompt-for-yes-or-no-in-bash
+# Slightly edited
+function confirm {
```

Comment:
This is the only time the `function` keyword is used. Why?
```suggestion
confirm () {
```